src/Frontend/reducers/user/userReducer.js

```diff
@@ -10,29 +10,42 @@
 
 import {
   CREATE_ACCOUNT,
+  CREATE_FAIL,
   ACTIVATE_ACCOUNT,
   REQUEST_API_KEY,
   LOGOUT
 }                                   from '../../constants/Portal';
 import { combineReducers }          from 'redux';
 
-// NOTE: each of these reducers is responsible for a certain sub-tree of the application's state.
-// The `registration`, `activation` and `authentication` reducers are specific for their feature (see component structure for more information),
-// but the `security` reducer contains security information about the application and can be composed to several user-specific actions.
+/*
+ NOTE: each of these reducers is responsible for a certain sub-tree of the application's state.
+ The `registration`, `activation` and `authentication` reducers are specific for their feature (see component structure for more information),
+ but the `security` reducer contains security information about the application and can be composed to several user-specific actions.
 
-// The app profile requires the following keys if initialized:
-// - `apiKey`
-// - `username`
-// - `isAdmin`
-// - `locale`
+ The app profile requires the following keys if initialized:
+ - `apiKey`
+ - `username`
+ - `isAdmin`
+ - `locale`
+ */
 const security = (state = { authenticated: false, appProfile: {} }, action) => {
   // TODO implement security related actions
   return state;
 };
 
 const registration = (state = { success: false, errors: {}, name_suggestions: [], id: null }, action) => {
   if (action.type === CREATE_ACCOUNT) {
-    // TODO implement state reduction for user creation
+    state = {
+      success:          action.payload.success,
+      errors:           action.payload.errors,
+      name_suggestions: action.payload.name_suggestions,
+      id:               action.payload.id
+    };
+  } else if (action.type === CREATE_FAIL) {
+    state.success          = action.payload.success;
+    state.name_suggestions = action.payload.name_suggestions;
+    state.errors           = action.payload.errors;
+    state.id               = action.payload.id;
```

Comment:
I'm not a big fan of these `if/else-if` constructs as the unnecessarily increase to code complexity. I'd prefer sth. like this (applying the early-return concept):

``` javascript
if (action.type === CREATE_ACCOUNT) {
  return Object.assign({
    {},
    state,
    { your values }
  });
}
if (action.type === CREATE_FAIL) {
  return Object.assign({
    // ...
  });
}
return state
```